editor/components/if-post-type-supports/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { get, some, castArray } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withAPIData, withConditionalRender } from '@wordpress/components';
+import { compose, getWrapperDisplayName } from '@wordpress/element';
+import { query } from '@wordpress/data';
+
+export function checkIfPostTypeSupports( postType, supportKeys ) {
+	return !! some(
+		castArray( supportKeys ), key => get( postType, [ 'data', 'supports', key ], false )
+	);
+}
+
+const applyQuery = query( ( select ) => ( {
+	postTypeSlug: select( 'core/editor', 'getCurrentPostType' ),
+} ) );
+
+const applyWithAPIData = withAPIData( ( { postTypeSlug } ) => {
+	return {
+		postType: postTypeSlug ? `/wp/v2/types/${ postTypeSlug }?context=edit` : undefined,
+	};
+} );
+
+const applyConditionalRender = ( supportKeys ) => withConditionalRender( ( { postType } ) => checkIfPostTypeSupports( postType, supportKeys ) );
+
+/**
+ * A Higher Order Component that provides a way to render a component or not
+ * depending on post type supports of the current post being edit.
+ *
+ * @param   {string|Array} supportKeys An array or string The component is rendered
+ *                                     if post type supports at least one of the supports keys passed.
+ * @returns {Component}                Wrapped component.
+ */
```

Comment:
Name should start with lowercase to match other HOCs. Why not start the name with ‘with’ instead of ‘if’?